handler/asset_allocations_handler.go

```diff
@@ -1,13 +1,127 @@
 package handler
 
 import (
+	"encoding/json"
+	"fmt"
 	"net/http"
 
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/vkhichar/asset-management/contract"
+	"github.com/vkhichar/asset-management/customerrors"
 	"github.com/vkhichar/asset-management/service"
 )
 
-func CreateAssetAllocationHandler(assetAllocation service.AssetAllocationService) http.HandlerFunc {
+func CreateAssetAllocationHandler(assetAllocationService service.AssetAllocationService) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+
+		assetId, err := uuid.Parse(vars["asset_id"])
+
+		if err != nil {
+			fmt.Printf("handler:incorrect asset id")
+			w.WriteHeader(http.StatusBadRequest)
+			responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "invalid asset id"})
+			w.Write(responseBytes)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+
+		var request contract.CreateAssetAllocationRequestForJson
+		claims := r.Context().Value("claims")
+		allocatedBy := claims.(*service.Claims).UserID
+
+		err = json.NewDecoder(r.Body).Decode(&request)
+		if err != nil {
+			fmt.Printf("handler: error while decoding request for creating maintenance activity for assets: %s", err.Error())
+
+			w.WriteHeader(http.StatusBadRequest)
+			responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "invalid request"})
+			w.Write(responseBytes)
+			return
+		}
+
+		req := contract.NewAssetAllocationRequest(assetId, allocatedBy, request)
+
+		assetAllocation, err := assetAllocationService.CreateAssetAllocation(r.Context(), req)
+
+		if err != nil {
+			if err == customerrors.UserNotExist {
+				fmt.Println("handler: user for this id does not exist")
+				w.WriteHeader(http.StatusBadRequest)
+				responseBytes, err := json.Marshal(contract.ErrorResponse{customerrors.UserNotExist.Error()})
```

Comment:
instead of json.Marshal(contract.ErrorResponse{customerrors.UserNotExist.Error()} you can simply do json.Marshal(contract.ErrorResponse{err.Error()}) as you are already checking err in if condition.